src/core/utils/extractErrorDetails.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * Extracts error details from a stack trace.
+ *
+ * @param {string} stack - The stack trace.
+ * @returns {Object} An object containing filename, lineNumber, and column.
+ */
+const extractErrorDetails = (stack) => {
+  const stackLines = (stack || "").split("\n");
+  const match = stackLines
+    .map((line) => {
+      const parts = line.match(/(?:\()?(.*?):(\d+):(\d+)(?:\))?/);
```

Comment:
The regex here is not totally clear to me. But I guess I have to see it in action 😅